jsx/Modal.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * This file contains the React Component for a Modal Window.
+ *
+ * @author Henri Rabalais
+ * @version 1.1.0
+ *
+ */
+import swal from 'sweetalert2';
+
+/**
+ * Modal Component.
+ * React wrapper for a Modal Window. Allows to dynamically toggle a Modal window.
+ *
+ * ================================================
+ * Usage:
+ * - Wrap the contents to be displayed by the Modal Window by the Modal Component.
+ * - Use the 'show' prop to toggle the Modal Component's presentation.
+ * - Use the 'title' prop to set a title for the Modal Component.
+ * - Use the 'onSubmit' prop set a submission function for the Modal's contents.
+ * - Use the 'throwWarning' prop to throw a warning upon closure of the Modal Window
+ * =================================================
+ *
+ */
+class Modal extends React.Component {
+  constructor() {
+    super();
+    this.closeModal = this.closeModal.bind(this);
+  }
+
+  closeModal() {
+    if (this.props.throwWarning) {
+      swal({
+        title: "Are You Sure?",
+        text: "Leaving the form will result in the loss of any information entered.",
+        type: "warning",
+        showCancelButton: true,
+        confirmButtonText: 'Proceed',
+        cancelButtonText: 'Cancel',
+      }).then(result => {
+        if (result.value) {
+          this.props.closeModal();
+        }
+      });
+    } else {
+      this.props.closeModal();
+    }
+  }
+
+  render() {
+    const headerStyle = {
```

Comment:
why not put this in with the CSS file?

Reply:
alternatively might be worth considering making these into default values and allowing for users to define their own CSS passed along via prop

Reply:
I've sort of been struggling with where to put styling with regard to react. I ended up with a bit of a mixed bag in this case because there didn't seem to be an easy way to do the css animation in react without adding third-party dependencies. I like the idea of a default with prop though.

Reply:
On second thought, for the sake of consistency I'm not sure if anyone should be changing these values. Is there anything in particular you feel needs to be overrideable/customizable?

Reply:
there may be cases where the CSS described here might look off in certain contexts or on certain displays. i wouldn't advocate for people to arbitrarily change the style but you never know when that extra flexibility might be needed

Reply:
Hmm, okay. If we are moving the styling to the css style sheet, do you suggest I simply allow the possibility of passing a title className to to this component?

Reply:
yep that sounds good to me